src/Sentry/Protocol/IJsonSerializable.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    internal interface IJsonSerializable
+    {
+        /// <summary>
+        /// Writes the object as JSON.
+        /// </summary>
+        void WriteTo(Utf8JsonWriter writer);
+    }
+
+    internal static class JsonSerializableExtensions
+    {
+        public static string ToJsonString(this IJsonSerializable serializable)
+        {
+            using var stream = new MemoryStream();
```

Comment:
This is just for testing, right?

Reply:
Currently yes, but I thought it would also be useful for debug logging (i.e. log the event content).